src/main/java/racingCarGame/domain/RacingCars.java

```diff
@@ -0,0 +1,74 @@
+package racingCarGame.domain;
+
+import racingCarGame.exception.DuplicateException;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class RacingCars {
+
+    private final List<Car> cars;
+    private final List<String> carsNames = new ArrayList<>();
+
+    private int position;
+
+    public RacingCars(List<Car> cars) throws DuplicateException {
+
+        checkDuplicate(cars);
+        this.cars = new ArrayList<>(cars);
+    }
+
+    private void checkDuplicate(List<Car> cars) throws DuplicateException {
+
+        for (Car car : cars) {
+            carsNames.add(car.getName());
+        }
+        Set<String> deduplicatedCar = new HashSet<>(carsNames);
+
+        if (cars.size() != deduplicatedCar.size()) {
+            throw new DuplicateException();
+        }
+    }
+
+    public int getNumberOfCars() {
+
+        return cars.size();
+    }
+
+    public void moveCar(String carName) {
+
+        for (Car car : cars) {
+            if (car.isEqual(carName)) {
+                car.movePosition();
+            }
+        }
```

Comment:
stream을 활용한다면 다음과 같이 적용할 수 있을 것 같아요

```suggestion
       cars.stream()
                .filter(car -> car.isEqual(carName))
                .forEach(car -> car.movePosition());
```